src/libertem_ui/live_plot.py

```diff
@@ -148,21 +148,47 @@ def is_multichannel(self):
     def _setup_multichannel(
         self,
         data: np.ndarray | dict[str, np.ndarray] | Sequence[np.ndarray],
-        dim: int = -1,
+        dim: int | tuple[int, ...] = -1,
     ):
         if isinstance(data, np.ndarray):
             if data.ndim == 2:
                 return data
             elif data.ndim == 3:
                 self._channel_data = data
-                self._channel_map = (dim % data.ndim)
+                assert isinstance(dim, int), "3D data requires int channel dim"
+                self._channel_map = (dim % data.ndim,)
                 slices = tuple(
-                    slice(None) if i != self._channel_map else 0
+                    slice(None) if i not in self._channel_map else 0
+                    for i in range(self._channel_data.ndim)
+                )
+                out_data = self._channel_data[slices]
+            elif data.ndim > 3:
```

Comment:
Any reason for _requiring_ an `int` in this case? Just thinking about the upstream code, the natural way of expressing this might be a tuple with a single element, or we would have to special case this basically on both sides.

Would something like `if isinstance(dim, int): dim = (dim,)` instead work, for keeping backwards-compat.?

Reply:
Good spot, there is no real backward compat to maintain but I think we can still accept `int` or `(int,)` for `ndim == 3`, and then reduce the special-casing of `ndim == 2` to just the insertion of the int into the tuple. I wrote this quite quickly so didn' t think about the homogeneity of it.